File: Niveau 2/2 – Découverte des tableaux/09 - Banquet municipal.py
nbPositionsTable = int(input())
nbChangementPosition = int(input())
pos = [0] * nbPositionsTable 
for i in range(nbPositionsTable):
    pos[i] = int(input())
for i in range(nbChangementPosition):
    p1 = int(input())
    p2 = int(input())
    temp = pos[p1]
    pos[p1] = pos[p2]
    pos[p2] = temp
for element in pos:
    print(element)

File: Niveau 2/2 – Découverte des tableaux/09 - Banquet municipal.cpp
#include <iostream>
using namespace std;
int main()
{
  int nbTotalPos, nbChangement;
  cin >> nbTotalPos >> nbChangement;
  int pos[nbTotalPos];
  for (int i = 0; i < nbTotalPos; i++)
  {
    cin >> pos[i];
  }
  for (int i = 0; i < nbChangement; i++)
  {
    int p1, p2;
    cin >> p1 >> p2;
    int temp = pos[p1];
    pos[p1] = pos[p2];
    pos[p2] = temp;
  }
  for (int i = 0; i < nbTotalPos; i++)
  {
    cout << pos[i] << endl;
  }
}